vmdb/app/views/ops/_ap_form_file.html.haml

```diff
@@ -25,37 +25,36 @@
           %td
-            = check_box_tag("entry_content", value="1", checked=false, :id=>"entry_content")
-      - if !session[:file_names].nil? && !session[:file_names].empty?
-        - session[:file_names].sort_by { |r| r["target"] }.each do |entry|
-          - if session[:edit_filename] != entry["target"]
-            %tr{:class => cycle('row0', 'row1'), :id => "#{entry["target"]}_tr"}
-              %td{:onclick => remote_function(:url=>{:action=>'ap_ce_delete', :file_name=>entry["target"], :id=>"#{@scan.id || "new"}"}), :title => _("Click to delete this entry")}
-                = image_tag("/images/toolbars/delete.png", :class=>"rollover small")
-              %td{:onclick => remote_function(:url=>{:action=>'ap_ce_select', :file_name=>entry["target"], :edit_entry=>'edit_file', :field=>"fname", :id=>"#{@scan.id || "new"}"}), :title => _("Click to update this entry")}
-                = h(entry["target"])
-              %td{:onclick => remote_function(:url=>{:action=>'ap_ce_select', :file_name=>entry["target"], :edit_entry=>'edit_file', :field=>"fname", :id=>"#{@scan.id || "new"}"}), :title => _("Click to update this entry")}
-                = h(entry["content"] ? entry["content"] : false)
-          - else
-            - if params[:add] != "new"
-              %tr{:class => cycle('row0', 'row1'), :id => "edit_tr"}
-                %td
-                  = image_submit_tag("/images/toolbars/import.png",
-                    :class => "rollover small",
-                    :id    => "accept",
-                    :name  => "accept",
-                    :alt   => _("Update this entry"),
-                    :title => _("Update this entry"),
-                    :id    => "#{@scan.id || 'new'}")
-                %td
-                  = text_field("entry", "fname", :maxlength=>MAX_NAME_LEN, "value"=>session[:edit_filename], :style => "width: 100%;")
-                  = hidden_field("item", "type1", :value=>"file")
-                %td
-                  - checked = entry["content"] ? true : false
-                  = check_box_tag("entry_content", value="1", checked=checked, :id=>"entry_content")
-                  = hidden_field("item", "type1", :value=>"file")
+            = check_box_tag("entry_content", 1, false, :id => "entry_content")
+      - Array(session[:file_names]).sort_by { |r| r["target"] }.each do |entry|
+        - if session[:edit_filename] != entry["target"]
+          %tr{ :class => cycle('row0', 'row1'), :id => "#{entry['target']}_tr" }
+            %td{ :onclick => remote_function(:url => {:action => 'ap_ce_delete', :file_name => entry["target"], :id => an_id}), :title => _("Click to delete this entry") }
+              = image_tag("/images/toolbars/delete.png", :class => "rollover small")
+            %td{ :onclick => remote_function(:url => {:action => 'ap_ce_select', :file_name => entry["target"], :edit_entry => 'edit_file', :field => "fname", :id => an_id}), :title => _("Click to update this entry") }
```

Comment:
@martinpovolny ahhhh, gotcha . . . maybe a better name would have been scan_id, but it's fine, thx.